src/interface/variable.cpp

```diff
@@ -158,6 +158,24 @@ std::string EdgeVariable<T>::info() {
   return s;
 }
 
+template <typename T>
+std::string ParticleVariable<T>::info() {
+  char tmp[100] = "";
+
+  // first add label
+  std::string s = this->label();
+  s.resize(20, '.');
+
+  // now append size
+  snprintf(tmp, sizeof(tmp), "%d", data.GetDim(1));
+  s += std::string(tmp);
+
+  // now append flag
+  s += " : " + this->metadata().MaskAsString();
+
+  return s;
+}
```

Comment:
This is very C-style of you... why not this?
```suggestion
std::string ParticleVariable<T>::info() {
  std::stringstream ss;

  // first add label
  std::string s = this->label();
  s.resize(20, '.');

  // combine
  ss << s << data.GetDim(1) << ":" << this->metadata().MaskAsString();

  return ss.str();
}
```

Reply:
Looks good to me -- I actually don't remember writing this, it's possible I copied this from an old version of `Variable`.